tests/simulationbase.py

```diff
@@ -39,8 +41,16 @@ def __init__(self, work_dir: Path) -> None:
     def init_provider(
             self,
             get_task_api_service: Callable[[Path], TaskApiService],
+            task_id
     ) -> None:
-        self.get_provider_task_api_service = get_task_api_service
+        self.mkdir_provider_task(task_id)
+        self._task_api_service = get_task_api_service(self.prov_task_work_dir)
+
+    @asynccontextmanager
+    async def start_provider(self) -> None:
```

Comment:
There's no much benefit in having this as a `asyncontextmanager` anymore

Reply:
removed